docs/literate/src/files/innards_of_the_basic_setup.jl

```diff
@@ -0,0 +1,244 @@
+# function from the OrdinaryDiffEq.jl package.
+
+ode = semidiscretize(semi, (0.0, 1.0));
+ 
+# The `semidiscretize` function involves a deep tree of recursive calls, with the primary ones
+# explained below.
+
+# - `allocate_coefficients(mesh, equations, solver, cache)`
+
+#   To apply initial conditions, a container needs to be generated to store the initial values of
+#   the target variables for each node within each element. The `allocate_coefficients` function
+#   initializes `u_ode` as a 1D zero-vector with a length that depends on the number of variables,
+#   elements, nodes, and dimensions. The use of a 1D vector format is consistent with the
+#   requirements of the `solve` function from OrdinaryDiffEq.jl. Therefore, Trixi.jl follows this
+#   format to be able to utilize the functionalities of OrdinaryDiffEq.jl.
+
+# - `wrap_array(u_ode, semi)`
+
+#   As previously noted, `u_ode` is constructed as a 1D vector to ensure compatibility with
+#   OrdinaryDiffEq.jl. However, for internal use within Trixi.jl, identification which part of the
+#   vector relates to specific variables, elements and nodes can be challenging.
+
+#   This is why the `u_ode` vector is wrapped by the `wrap_array` function to create a
+#   multidimensional array `u`, with each dimension representing variables, nodes and elements.
+#   Consequently, navigation within this multidimensional array becomes noticeably easier.
+
+#   "Wrapping" in this context involves the creation of a reference to the same storage location
+#   but with an alternative structural representation. This approach enables the use of both
+#   instances `u` and `u_ode` as needed, so that changes are simultaneously reflected in both.
+#   This is possible because, from a storage perspective, they share the same stored data, while
+#   access to this data is provided in different ways.
+
+# - `compute_coefficients!(u, initial_conditions, t, mesh::DG, equations, solver, cache)`
+
+#   Now the variable `u`, intended to store solutions, has been allocated and wrapped, it is time
+#   to apply the initial conditions. The `compute_coefficients!` function calculates the initial
+#   conditions for each variable at every node within each element and properly stores them in the
+#   `u` array.
+
+# At this stage, the `semidiscretize` function has all the necessary components to initialize and
+# return an `ODEProblem` object, which will be used by the `solve` function to compute the
+# solution.
+
+# In summary, the internal workings of `semidiscretize` with brief descriptions can be presented
+# as follows.
+
+# ![semidiscretize_structure](https://github.com/trixi-framework/Trixi.jl/assets/119304909/491eddc4-aadb-4e29-8c76-a7c821d0674e)
+
+
+# ## Functions `solve` and `rhs!`
+
+# Once the `ODEProblem` object is initialized, the `solve` function from the OrdinaryDiffEq.jl
+# package can be utilized to compute an approximated solution using the instructions contained in
+# the `ODEProblem` object.
```

Comment:
I would state here that we also use the **solver** form `OrdinaryDiffEq.jl`  to avoid potential confusion with the custom integrators from Trixi. To me, it was initially not clear that also the ODE-solver itself comes from `OrdinaryDiffEq.jl` .

Reply:
Good point! Thanks, I recognized that I used solve function in meaning of ODE-solvers sometimes, now should be OK.